src/org/apache/camel/idea/CamelJavaReferenceContributor.java

```diff
@@ -16,137 +16,22 @@
-            if (element instanceof PsiMethod) {
-                return handleElementRename(((PsiMethod) element).getName());
-            }
-            return super.bindToElement(element);
-        }
-
-        @Nullable
-        public PsiElement resolve() {
-            return getElement();
-        }
-
-        @NotNull
-        public Object[] getVariants() {
-            List<Object> answer = null;
-
-            // special IDEA hack which is really needed (yes they do this)
-            String val = getValue();
-            int hackIndex = val.indexOf(CompletionUtil.DUMMY_IDENTIFIER);
-            if (hackIndex == -1) {
-                hackIndex = val.indexOf(CompletionUtil.DUMMY_IDENTIFIER_TRIMMED);
-            }
-            if (hackIndex > -1) {
-                val = val.substring(0, hackIndex);
-            }
-
-            // is this a possible Camel endpoint uri which we know
-            String componentName = StringUtils.asComponentName(val);
-            if (componentName != null && camelCatalog.findComponentNames().contains(componentName)) {
-
-                // it is a known Camel component
-                String json = camelCatalog.componentJSonSchema(componentName);
-
-                // gather list of names
-                List<Map<String, String>> rows = JSonSchemaHelper.parseJsonSchema("properties", json, true);
-
-                // grab all existing parameters
-                String query = val;
-                // strip up ending incomplete parameter
-                if (query.endsWith("&") || query.endsWith("?")) {
-                    query = query.substring(0, query.length() - 1);
-                }
-
-                Map<String, String> existing = null;
-                try {
-                    existing = camelCatalog.endpointProperties(query);
-                } catch (Exception e) {
-                    // ignore
-                }
-
-                // are we editing an existing parameter value
-                // or are we having a list of suggested parameters to choose among
-                boolean editSingle = val.endsWith("=");
-                if (editSingle) {
-                    // parameter name is before = and & or ?
-                    int pos = Math.max(val.lastIndexOf('&'), val.lastIndexOf('?'));
-                    String name = val.substring(pos + 1);
-                    name = name.substring(0, name.length() - 1); // remove =
-                    answer = addSmartCompletionForSingleValue(val, rows, name);
-                } else {
-                    answer = addSmartCompletionSuggestions(val, rows, existing);
-                }
-            }
-            if (answer != null) {
-                return answer.toArray();
-            } else {
-                return Collections.emptyList().toArray();
-            }
-        }
+public class CamelJavaReferenceContributor extends CamelContributor {
+
+    public CamelJavaReferenceContributor() {
+        extend(CompletionType.BASIC,
```

Comment:
Actually I tried the first approach with PsiLiteral, but that didn't work for some reason. But I limit it by checking if psiElement().inside(PsiClass.class)